ArduCopter/radio.cpp

```diff
@@ -83,9 +83,10 @@ void Copter::read_radio()
     const uint32_t tnow_ms = millis();
 
     if (rc().read_input()) {
-        ap.new_radio_frame = true;
-
         set_throttle_and_failsafe(channel_throttle->get_radio_in());
+        if (!rc().has_valid_input()) {
```

Comment:
this part is a little surprising.  We set new_radio_frame = true when we **don't** have valid input.

Reply:
Apologies, fixed.  I tried various different flavours of this.

Waiting for CI to pass before writing an autotest for this :-)  Or. more probably, testing in SITL as detecting this via mavlink may be problematic.